src/classes/VOL_CTRL_PersonalSiteContactInfo.cls

```diff
@@ -258,198 +269,223 @@ global with sharing class VOL_CTRL_PersonalSiteContactInfo {
     // Note that it stores the formatted Time in the Hours' System_Note__c field (in memory only).
     // Note that it stores the formatted Date in the Hours' Comment field (in memory only).
     private void dateTimeFixup(list<Volunteer_Hours__c> listHours) {
-    	
-    	// get default time zone for site guest user
-    	User u = [Select TimeZoneSidKey From User where id =: Userinfo.getUserId()];
-    	
-    	// javascript formatting used 'tt' for am/pm, whereas apex formatting uses 'a'.
-    	string strFormat = strDateFormat + ' ' + strTimeFormat.replace('tt','a');
-    	string strFormatEndTime = strTimeFormat.replace('tt','a');
-    	
-    	for (Volunteer_Hours__c hr : listHours) {
-			string strTimeZone = hr.Volunteer_Job__r.Volunteer_Website_Time_Zone__c;
-			if (strTimeZone == null) strTimeZone = hr.Volunteer_Job__r.Campaign__r.Volunteer_Website_Time_Zone__c;
-			if (strTimeZone == null) strTimeZone = u.TimeZoneSidKey;
-			
-			DateTime dtStart = hr.Planned_Start_Date_Time__c == null ? hr.Shift_Start_Date_Time__c : hr.Planned_Start_Date_Time__c;
-			if (dtStart == null) {
-				// if we don't have a start time, then format by GMT, to avoid timezones moving the date!
-				dtStart = hr.Start_Date__c;
-				hr.Comments__c = dtStart.formatGmt(strDateFormat);
-			} else {
-				double duration = hr.Hours_Worked__c == null ? hr.Volunteer_Shift__r.Duration__c : hr.Hours_Worked__c;
-				DateTime dtEnd = dtStart.addMinutes(integer.valueOf(duration * 60));
-				string strStart = dtStart.format(strFormat, strTimeZone);
-				
-				// see if start and end are on the same day
-				if (dtStart.format('d', strTimeZone) == dtEnd.format('d', strTimeZone)) {
-					hr.System_Note__c =  dtStart.format(strFormatEndTime, strTimeZone) + ' - ' + dtEnd.format(strFormatEndTime, strTimeZone);	
-				} else {
-					hr.System_Note__c =  dtStart.format(strFormatEndTime, strTimeZone) + ' - ' + dtEnd.format(strFormat, strTimeZone);	    				
-				} 
-				
-				// also save user formated Start Date in Comments field
-				hr.Comments__c = dtStart.format(strDateFormat, strTimeZone);
-			}  			
-    	}
+        try {
+            // Ensure the user has access to the object and fields before querying
+            UTIL_Describe.checkReadAccess('User', new Set<String>{'TimeZoneSidKey'});    
```

Comment:
Just tested this in my org, and indeed the Sites Guest User profile does not have permissions on User.  What's worse, there is no way to set it from Salesforce setup!  I'm not really sure what controls permissions to the user object.  the only thing I see under setup for any profile choosing Users is the recordTypes and page Layouts.  but there is no list of fields displayed to enabled/disable for read and edit.  So let's remove any checks to User in all the classes (I think there were just a few places).